File: runtime/src/test/java/com/siimkinks/sqlitemagic/ComplexColumnTestFixtures.java
package com.siimkinks.sqlitemagic;

import androidx.annotation.NonNull;

import static com.siimkinks.sqlitemagic.Utils.LONG_PARSER;

final class ComplexColumnTestFixtures {
  static final Table<Article> ARTICLE = new Table<>("article", null, 2);
  static final AccountNumericColumn ACCOUNT = new AccountNumericColumn();

  private ComplexColumnTestFixtures() {
  }

  static final class Article {
  }

  record AccountId(long value) {
  }

  static final class AccountNumericColumn extends ComplexNumericColumn<AccountId, AccountId, Number, Article, NotNullable> {
    AccountNumericColumn() {
      super(ARTICLE, "account", false, LONG_PARSER, false, null);
    }

    @NonNull
    @Override
    String toSqlArg(AccountId value) {
      return Long.toString(value.value + 1L);
    }
  }
}

File: runtime/src/test/kotlin/com/siimkinks/sqlitemagic/ComplexColumnTestFixtures.kt
package com.siimkinks.sqlitemagic

import com.siimkinks.sqlitemagic.Utils.LONG_PARSER

internal object ComplexColumnTestFixtures {
  val ARTICLE: Table<Article> = testTable(
    name = "article",
    alias = null,
    nrOfColumns = 2
  )
  val ACCOUNT = AccountNumericColumn()

  class Article

  data class AccountId(val value: Long)

  class AccountNumericColumn : ComplexNumericColumn<AccountId, AccountId, Number, Article, NotNullable>(
    table = ARTICLE,
    name = "account",
    valueParser = LONG_PARSER,
    nullable = false,
    alias = null
  ) {
    override fun toSqlArg(value: AccountId) = (value.value + 1L).toString()
  }
}
